components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/ClaimMetadataManagementServiceImpl.java

```diff
@@ -108,7 +118,8 @@ public void renameClaimDialect(ClaimDialect oldClaimDialect, ClaimDialect newCla
     public void removeClaimDialect(ClaimDialect claimDialect, String tenantDomain) throws ClaimMetadataException {
 
         if (claimDialect == null || StringUtils.isBlank(claimDialect.getClaimDialectURI())) {
-            throw new ClaimMetadataException("Claim dialect URI cannot be empty");
+            throw new ClaimMetadataClientException(ERROR_CODE_EMPTY_CLAIM_DIALECT.getCode(),
+                    "Claim dialect URI cannot be empty");
```

Comment:
This error message is almost similar to the "ERROR_CODE_EMPTY_CLAIM_DIALECT" error message hence did not add a new error message in the enum.